generators/browser-compatible-base/src/ast/AbstractWriter.ts

```diff
@@ -150,6 +150,24 @@ export class AbstractWriter {
         this.indentLevel--;
     }
 
+    public delimit<T extends AbstractAstNode>(
+        nodes: T[],
+        delimiter: string,
+        writeFunction: (node: T) => void,
+        prefix: string = "",
```

Comment:
Let's use the in-lined request params I mentioned in an [earlier comment.](https://github.com/fern-api/fern/pull/5422#discussion_r1887542602). Otherwise the call-sites are difficult to follow.

Reply:
I like dropping the prefix/suffix, but I think we could probably drop the proposed newline arg. The user can supply writeLine for the write function, which would have the desired effect.